### ISSUE TITLE
Add tests for add products task's product types

### ISSUE THREAD

**Issue by ilyasfoo** [woothemes/woocommerce#49653]
### Issue Description

Follow-up to https://github.com/woocommerce/woocommerce/pull/49631.

We should add unit or e2e tests that covers adding product with all the types available in `Add product` task.

### Acceptance criteria

1. The test should test creating product with the correct product types from options available in `Add product` task

**Comment by timmy5685** [comment]
Coming back to this during a triage pass. It looks like most of it is done but I'd like engineering feedback to confirm and decide if we should finish the remaining narrow piece of work that seems to still be open. Here were the results of my research with claude:

Already covered: the products fill unit tests (`products/test/index.tsx`) click each product-type option and assert the correct type flows through — e.g. "Digital product" → `tasklist_product_template_selection` with `{ product_type: 'digital' }`, "Grouped product" → `{ product_type: 'grouped' }` — plus tests that the right types render per profile. So the options and their per-click type wiring are covered.

Still missing — and it's exactly the #49631 surface: those tests assert on tracks events, not on the actual product creation. The create-mapping (`use-create-product-by-type.ts` — `getProductEditPageLink` → `createProductFromTemplate( type )`) has no test. So we verify that clicking "Variable" *emits* `product_type: 'variable'`, but nothing verifies it actually *creates* a variable product — which is what #49631 fixed and what this was filed to guard.

Re-scoping to that one concrete piece: a unit test for `use-create-product-by-type.ts` asserting each type maps to the correct `createProductFromTemplate` call. Small and well-defined, rather than the open-ended "add tests" it reads as today.

**Comment by vladolaru** [comment]
I agree with the scoped triage above. This still looks valid, but it should be kept narrow.

The original bug from [#49218](https://github.com/woocommerce/woocommerce/issues/49218) was fixed in [#49631](https://github.com/woocommerce/woocommerce/pull/49631) by simplifying the Add products task so the selected product type is passed through as the template name. Current trunk still has that behavior in [`use-create-product-by-type.ts`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/admin/client/task-lists/fills/products/use-create-product-by-type.ts#L20-L50): it calls `createProductFromTemplate( { template_name: type, status: 'draft' }, { _fields: [ 'id' ] } )`, then navigates to the edit screen with `tutorial_type=${ type }`.

The remaining gap is that the product-task tests mock `useCreateProductByType`, so they cover rendering and click/Tracks wiring, but not the actual creation request or navigation URL from the hook. The five template-backed frontend keys are still `physical`, `digital`, `variable`, `grouped`, and `external` in [`constants.tsx`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/admin/client/task-lists/fills/products/constants.tsx#L21-L66), and matching CSV templates exist for those keys in the onboarding API templates directory.

My recommendation: keep this open, but scope it to a small JS unit test for `useCreateProductByType`. The test should assert each of the five product type keys calls:

```ts
createProductFromTemplate(
	{ template_name: type, status: 'draft' },
	{ _fields: [ 'id' ] }
);
```

It should also assert the resulting navigation URL includes `wc_onboarding_active_task=products`, `tutorial=true`, and `tutorial_type=${ type }`.

I would avoid turning this into broad E2E coverage unless the owning team specifically wants that extra cost. The regression from #49631 is at the hook mapping boundary, so a focused unit test should be enough to close this.
